src/event/log_event.rs

```diff
@@ -177,3 +177,58 @@ impl Serialize for LogEvent {
         serializer.collect_map(self.fields.iter())
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use std::{fs, io::Read, path::Path};
+
+    fn parse_artifact(path: impl AsRef<Path>) -> std::io::Result<Vec<u8>> {
+        let mut test_file = match fs::File::open(path) {
+            Ok(file) => file,
+            Err(e) => Err(e)?,
+        };
+
+        let mut buf = Vec::new();
+        test_file.read_to_end(&mut buf)?;
+
+        Ok(buf)
+    }
+
+    // This test iterates over the `tests/data/fixtures/log_event` folder and:
+    //   * Ensures the EventLog parsed from bytes and turned into a serde_json::Value are equal to the
+    //     item being just plain parsed as json.
+    //
+    // Basically: This test makes sure we aren't mutilating any content users might be sending.
+    #[test]
+    fn json_value_to_vector_log_event_to_json_value() {
+        crate::test_util::trace_init();
+        const FIXTURE_ROOT: &str = "tests/data/fixtures/log_event";
+
+        tracing::trace!(?FIXTURE_ROOT, "Opening.");
```

Comment:
nitpick: Any reason for `tracing::trace!` here instead of just `trace!`? Everywhere else we use the latter.